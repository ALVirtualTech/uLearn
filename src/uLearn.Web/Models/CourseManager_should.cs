using System;
using System.Collections.Generic;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace uLearn.Web.Models
{
	[TestFixture]
	public class CourseManager_should
	{
		[Test, UseReporter(typeof(DiffReporter))]
		public void load_all_courses()
		{
			var manager = WebCourseManager.Instance;
			var courses = manager.GetCourses().ToList();
			foreach (var c in courses)
			{
				Console.WriteLine("Course: " + c.Id + " " + c.Title);
				foreach (var s in c.Slides)
					Console.WriteLine(s.Info.UnitName + " " + s.Title);
				Console.WriteLine();
			}
			Approvals.VerifyAll(courses.Where(c => !c.Id.StartsWith("BasicProgramming")), "Courses");
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void load_ForTestsCourse_slides()
		{
			var manager = WebCourseManager.Instance;
			Approvals.VerifyAll(manager.GetCourse("ForTests").Slides, "Slides");
			
		}
	}
}
