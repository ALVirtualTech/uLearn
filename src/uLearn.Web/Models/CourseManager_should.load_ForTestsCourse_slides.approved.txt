Slides[0] = Title: S010_TextBlocks, Id: {108C89D9-36F0-45E3-BBEE-B93AC971063F}, MaxScore: 0
Slides[1] = Title: S011_GoodTeX, Id: {118C89D9-36F0-45E3-BBEE-B93AC971063F}, MaxScore: 0
Slides[2] = Title: S012_WrongTeX, Id: {128C89D9-36F0-45E3-BBEE-B93AC971063F}, MaxScore: 0
Slides[3] = Title: S013_VideoBlock, Id: {138C89D9-36F0-45E3-BBEE-B93AC971063F}, MaxScore: 0
Slides[4] = Title: S014_CodeBlock, Id: {148C89D9-36F0-45E3-BBEE-B93AC971063F}, MaxScore: 0
Slides[5] = ExerciseSlide: Exercise: public static void Print()
{
}, Hints: System.Collections.Generic.List`1[System.String]
Slides[6] = Title: Шаблонный тест, Id: {42E607B8-E7D8-4D96-B3B6-B0CDED8BA8A8}, MaxScore: 3
